### ISSUE TITLE
Missing 'All classes' link on device with multiple facilities

### ISSUE THREAD

**Issue by pcenov** [learningequality/kolibri#10673]
## Observed behavior
The 'All classes' link is missing on a device with multiple facilities so once I have selected one of the available classes I have to navigate to a different plugin in order to go back to the Classes page.

## Expected behavior
The  'All classes' link should be displayed in that case.

![2023-05-10_16-52-00](https://github.com/learningequality/kolibri/assets/79847249/d486d842-6c45-4d3f-8124-1ae98529e0f7)

general expectations for `Coach` behavior & UI:

- if **user 1** is a multi-facility admin of Facility A & Facility B, and Facility A has only one class, `Class home` goes directly to that class's page and when there, the back link is "← Change learning facility"
- when **user 1** selects Facility B, which has multiple classes, they are redirected to the `All Classes` page which has a back link to "← Change learning facility".
    - selecting Class B1 will take them to its specific class page, which contains a back link to "← All classes"
- if **user 2** has access to only Facility A, which has one class (or user has other limitations only allowing access to one class), `Class home` goes directly to that class's page and there is no back link displayed.
- if **user 3** has access to only Facility B, which has multiple classes, they are redirected to the `All Classes` page and there is no back link displayed.
   - selecting Class B1 will take you to its specific class page with a back link to "← All classes"

## Steps to reproduce the issue
1. Install the the build asset from [this PR](https://github.com/learningequality/kolibri/pull/10550)
2. Import a facility with multiple classes
3. Navigate to Coach > Class home and select a class
4. Attempt to go back to the Classes page

Note: observed in edge case when multi-facility user's main facility has 1 class and imported facility has multiple, detailed in comments

## Video

https://github.com/learningequality/kolibri/assets/79847249/a1d02b87-1739-4c01-83a2-a831a1341164

## Usage Details
Ubuntu 22  - Chrome, Firefox

**Comment by pcenov** [comment]
@radinamatic

**Comment by thanksameeelian** [comment]
i'm actually unable to recreate this problem because another bug is keeping me from being able to see this view at all: #10643. when i initially mentioned the linked bug, which i encounter every time i try to navigate within `Coach` > `Class Home` without fail, @marcellamaki mentioned she believed it was related to routing work that was still ongoing within `Coach`. 

~@pcenov, have you ever encountered the bug detailed in #10643 (& in the screen recording below)? because i consistently encounter it, i'd consider this issue "blocked" but now i'm curious if i'm the only one experiencing it! 🙂~ (please see update below!)

https://github.com/learningequality/kolibri/assets/43046460/0b49b4f0-2722-4766-801b-b7718a59cee9

**Comment by thanksameeelian** [comment]
following up on my comment after a conversation with @rtibbles sorted things out somewhat: it appears that i'm experiencing the bug in #10643 & the screen recording because all of the classes i'm attempting to access contain quizzes. 

in the examples used to illustrate this "missing 'all classes' link" issue, it appears that the classes don't contain quizzes, so navigation to them isn't triggering the same bug, which seems to be occurring within a code block that is never accessed if no quizzes exist. 

no action or anything needed, i just wanted to clarify my earlier comment! i'll be working on this but likely only once #10643 is addressed, which should be quite soon.

**Comment by thanksameeelian** [comment]
@pcenov i haven't been able to exactly replicate the issue reported here, as throughout my navigation within classes in `Coach` as a multi-facility user i've been consistently seeing the "← All classes" back links as expected if there are multiple classes. but there is some behavior proposed in #10650 that i think aligns with the sentiment of this issue and could be addressed in response to this issue instead:

> * change backlink text from "← All Facilities" to "← Change learning facility" (in `commonCoreStrings`)
> * add backlink to all main Coach subtopic pages, not just "Class home"

(changing the text to align with the recent update to the same type of link in the `Facility` plugin)

in talking through this issue with @rtibbles, the following rules were laid out to help govern navigation behavior in `Coach` and establish expectations:

* if **user 1** is a multi-facility admin of Facility A & Facility B, and Facility A has only one class, `Class home` goes directly to that class's page and when there, the back link is "← Change learning facility"
* when **user 1** selects Facility B, which has multiple classes, they are redirected to the `All Classes` page which has a back link to "← Change learning facility". 
    * selecting Class B1 will take them to its specific class page, which contains a back link to "← All classes"
* if **user 2** has access to only Facility A, which has one class (or user has other limitations only allowing access to one class), `Class home` goes directly to that class's page and there is no back link displayed.
* if **user 3** has access to only Facility B, which has multiple classes, they are redirected to the `All Classes` page and there is no back link displayed.
     * selecting Class B1 will take you to its specific class page with a back link to "← All classes"
     
i was planning on adding these expectations to the description of this issue (and removing the related bullet points from #10650), because it seemed like the specific issue you raised, even if i wasn't able to recreate it exactly, should be appropriately handled by a PR that enforces these expectations throughout `Coach`. 

i wanted to check in with you before editing though, in case you felt like the case you're experiencing isn't adequately covered by this, you feel my interpretation is incorrect, or you have any other objections to potentially expanding the scope of this issue slightly.

**Comment by pcenov** [comment]
Hi @thanksameeelian, thank you for your detailed investigation on this one and yes, I believe the use cases you've outlined cover nicely the expected behavior for `Coach`.

I was also able to identify that the issue I've reported is only valid if there's a single class in my Main facility and there are multiple classes in the imported facility. I also noticed that if I use the browser's Back and Forward buttons then the `All classes` link gets displayed. Here's a video of that:

https://github.com/learningequality/kolibri/assets/79847249/521a22ff-c2d0-49a6-bf05-b5ef8a3b9431
